### PR TITLE
More updates to setup script experience

### PR DESCRIPTION
**Pull request by jkamdjou**
How to test (s/o @hugh for this):

`curl -sL https://raw.githubusercontent.com/sublime-security/sublime-platform/josh.setup_script/install-and-launch.sh | remote_branch=josh.setup_script bash`

Note that Github caches the raw content and [only updates every `5 minutes](https://stackoverflow.com/questions/46551413/github-not-update-raw-after-commit). There doesn't seem to be a way to bypass this, except possibly by requesting the commit itself instead of the branch.

Local testing: `clone_platform=false bash install-and-launch.sh`

image

image

### CODE REVIEW

**Review by ianthiel — APPROVED** [review]

**Comment by jkamdjou** [comment]
@hugh-sublime i got this merged today so i could run through an on boarding with someone, should be pretty straight forward but take a look when you can to be sure everything looks good 🙏

File: install-and-launch.sh [hunk 1]
```diff
@@ -127,16 +147,17 @@ fi
...
-    print_error "Failed to launch Sublime Platform"
-    echo "See https://docs.sublimesecurity.com/docs/quickstart-docker#troubleshooting for troubleshooting tips"
-    echo "If you'd like to reinstall Sublime then follow the steps outline in https://docs.sublimesecurity.com/docs/quickstart-docker#wipe-postgres-volume"
-    echo "Afterwards, run: rm -rf ./sublime-platform"
-    echo "You can then go through the Sublime Platform installation again"
+    print_error "Failed to launch Sublime Platform\n"
+    printf "Troubleshooting tips: https://docs.sublimesecurity.com/docs/quickstart-docker#troubleshooting\n\n"
+    printf "If you'd like to re-install Sublime then follow these steps: https://docs.sublimesecurity.com/docs/quickstart-docker#wipe-your-data\n\n"
+    printf "Afterwards, run: rm -rf ./sublime-platform\n\n"
+    printf "You can then go through the Sublime Platform installation again"
```
Comment (hugh-sublime) [inline install-and-launch.sh:154]: You'll probably want a newline at the end of this one.

File: preflight_checks.sh [hunk 2]
```diff
@@ -45,13 +46,14 @@ case "$(uname -s | tr '[:upper:]' '[:lower:]')" in
 esac
 
 if [ -z "$machine" ]; then
-    print_warning "Warning: You are using a non-recommended operating system so subsequent failures may occur"
+    print_warning "Warning: You are using a non-recommended operating system so subsequent failures may occur\n\n"
+    printf "Recommended operating systems: https://docs.sublimesecurity.com/docs/quickstart-docker#requirements"
```
Comment (hugh-sublime) [inline preflight_checks.sh:50]: Newline or use `echo`?

File: preflight_checks.sh [hunk 3]
```diff
@@ -139,39 +142,42 @@ docker_compose_version=${docker_compose_version##*version v}
 
 if [ -z "$docker_compose_version" ]; then
     if [ "$machine" == "linux" ]; then
-        echo "docker compose not installed. Please install docker compose and retry (https://docs.docker.com/compose/install/linux/#install-using-the-repository)"
+        print_error "Docker Compose is not installed\n\n"
+        printf "Please install Docker Compose and retry: https://docs.docker.com/compose/install/linux/#install-using-the-repository"
```
Comment (hugh-sublime) [inline preflight_checks.sh:146]: Newline or `echo`?

File: preflight_checks.sh [hunk 4]
```diff
@@ -139,39 +142,42 @@ docker_compose_version=${docker_compose_version##*version v}
...
         exit 1
     fi
 
-    echo "docker compose not installed. Please install docker compose and retry (https://docs.docker.com/compose/install/)"
+    print_error "Docker Compose is not installed\n\n"
+    printf "Please install Docker Compose and retry: https://docs.docker.com/compose/install/"
```
Comment (hugh-sublime) [inline preflight_checks.sh:151]: Same as above.

File: preflight_checks.sh [hunk 5]
```diff
@@ -139,39 +142,42 @@ docker_compose_version=${docker_compose_version##*version v}
...
     exit 1
 fi
 
 if version_lt "$(major_minor "$docker_compose_version")" "2.4"; then
-    echo "docker compose version $docker_compose_version does not meet the minimum version of 2.4. Please update docker compose and retry"
+    print_error "Docker Compose version $docker_compose_version does not meet the minimum version of 2.4\n\n"
+    printf "Please update Docker Compose and retry"
```
Comment (hugh-sublime) [inline preflight_checks.sh:157]: Same

File: preflight_checks.sh [hunk 6]
```diff
@@ -139,39 +142,42 @@ docker_compose_version=${docker_compose_version##*version v}
...
     exit 1
 fi
 
 if [ "$auto_updates" == "true" ] && ! which cron > /dev/null 2>&1; then
-    echo "cron not installed. Please install cron and retry"
-    exit 1
+    print_error "Cron is not installed\n\n"
+    printf "Please install cron and retry"
```
Comment (hugh-sublime) [inline preflight_checks.sh:163]: Same

File: preflight_checks.sh [hunk 7]
```diff
@@ -139,39 +142,42 @@ docker_compose_version=${docker_compose_version##*version v}
...
 # docker without realizing they could loose data (from our platform or other applications).
 if which snap > /dev/null 2>&1 && snap list | grep -i docker > /dev/null 2>&1; then
-    echo "snap versions of docker software detected! Cannot proceed."
-    echo "snap versions of docker are not recommended and can cause issues when using compose in the future (e.g. cannot bring containers down)"
-    echo "please uninstall snap docker packages and install with apt"
-    echo "If you have existing docker containers or volumes or are otherwise unsure, please contact support@sublimesecurity.com for assistance"
+    print_error "Snap versions of Docker detected! Cannot proceed\n"
+    print_error "Snap versions of Docker are not recommended because they can cause issues when using Docker Compose in the future (e.g. cannot bring containers down)\n\n"
+    printf "Please uninstall Snap Docker packages and install with apt instead.\n\n"
+    printf "If you have existing docker containers or volumes or are otherwise unsure, please contact support@sublimesecurity.com for assistance"
```
Comment (hugh-sublime) [inline preflight_checks.sh:179]: Same

**Review by hugh-sublime — COMMENTED** [review]
For some reason the original `@hugh` tag didn't message me in Slack but `@hugh-sublime` tagged me properly. Everything seems good but we may want to add some newlines in the places we're using `printf` instead of `echo`.

**Comment by jkamdjou** [comment]
> For some reason the original `@hugh` tag didn't message me in Slack but `@hugh-sublime` tagged me properly. Everything seems good but we may want to add some newlines in the places we're using `printf` instead of `echo`.

i think that original mis tag was my fault. 🙏 thank you! will update

**Pull request merged**